popos.py

```diff
@@ -9,22 +9,39 @@ def __init__(self, type):
 
 
 class Book:
-	def __init__(self, object):
-		self.title = object['volumeInfo'].setdefault('title', '')
-		self.description = object['volumeInfo'].setdefault('description', '')
-		self.published_date = object['volumeInfo'].setdefault('publishedDate', '')
+	def __init__(self, title, description, published_date, image_url, isbn, authors, genres):
+		self.title = title
+		self.description = description
+		self.published_date = published_date
+		self.isbn = isbn
+		self.image_url = image_url
+		self.authors = authors
+		self.genres = genres
 
-		if 'imageLinks' in object['volumeInfo']:
-			self.image_url = object['volumeInfo']['imageLinks'].setdefault('thumbnail', '')
+	@classmethod
+	def from_google_response(cls, payload):
+		book_data = payload['volumeInfo']
+
+		title = book_data.setdefault('title', '')
+		description = book_data.setdefault('description', '')
+		published_date = book_data.setdefault('publishedDate', '')
+
+		if 'industryIdentifiers' in book_data:
+			isbn = book_data['industryIdentifiers'][0].setdefault('identifier', '')
 		else:
-			self.image_url = ''
+			isbn = ''
+
+		if 'imageLinks' in book_data:
+			image_url = book_data['imageLinks'].setdefault('thumbnail', '')
+		else:
+			image_url = ''
+
+		authors = []
+		for author in book_data.setdefault('authors', []):
+			authors.append(Author(name=author))
 
-		self.author_list = object['volumeInfo'].setdefault('authors', [])
-		self.authors = []
-		for author in self.author_list:
-			self.authors.append(Author(name=author))
+		genres = []
+		for genre in book_data.setdefault('categories', []):
+			genres.append(Genre(type=genre))
 
-		self.genre_list = object['volumeInfo'].setdefault('categories', [])
-		self.genres = []
-		for genre in self.genre_list:
-			self.genres.append(Genre(type=genre))
\ No newline at end of file
+		return cls(title, description, published_date, image_url, isbn, authors, genres)
```

Comment:
This constructor should now be used. `Book.from_google_response(payload)` instead of `Book(payload)`

Reply:
If I'm not mistaken, passing the payload into ```from_google_response``` instead of directly into ```Book``` can also allow future flexibility if we decide to use a different API other than Google Books, correct? If we choose to create a book object with information returned from another API, we could define another method for it.

Reply:
@rlew421 Yes, exactly!